test/src/protocol/RemotingCommandTest.cpp

```diff
@@ -0,0 +1,285 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdlib.h>
+#include <iostream>
+#include <memory>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+#include "CommandHeader.h"
+#include "MQProtos.h"
+#include "MQVersion.h"
+#include "MemoryOutputStream.h"
+#include "RemotingCommand.h"
+#include "dataBlock.h"
+
+using std::shared_ptr;
+
+using ::testing::InitGoogleMock;
+using ::testing::InitGoogleTest;
+using testing::Return;
+
+using rocketmq::CommandHeader;
+using rocketmq::GetConsumerRunningInfoRequestHeader;
+using rocketmq::GetEarliestMsgStoretimeResponseHeader;
+using rocketmq::GetMaxOffsetResponseHeader;
+using rocketmq::GetMinOffsetResponseHeader;
+using rocketmq::GetRouteInfoRequestHeader;
+using rocketmq::MemoryBlock;
+using rocketmq::MemoryOutputStream;
+using rocketmq::MQRequestCode;
+using rocketmq::MQVersion;
+using rocketmq::NotifyConsumerIdsChangedRequestHeader;
+using rocketmq::PullMessageResponseHeader;
+using rocketmq::QueryConsumerOffsetResponseHeader;
+using rocketmq::RemotingCommand;
+using rocketmq::ResetOffsetRequestHeader;
+using rocketmq::SearchOffsetResponseHeader;
+using rocketmq::SendMessageResponseHeader;
+
+TEST(remotingCommand, init) {
+    RemotingCommand remotingCommand;
+    EXPECT_EQ(remotingCommand.getCode(), 0);
+    EXPECT_EQ(remotingCommand.getOpaque(), 0);
+    EXPECT_EQ(remotingCommand.getRemark(), "");
+    EXPECT_EQ(remotingCommand.getVersion(), 0);
+    //  EXPECT_EQ(remotingCommand.getFlag() , 0);
+    EXPECT_EQ(remotingCommand.getMsgBody(), "");
+    EXPECT_TRUE(remotingCommand.getCommandHeader() == nullptr);
+
+    RemotingCommand twoRemotingCommand(13);
+    EXPECT_EQ(twoRemotingCommand.getCode(), 13);
+    EXPECT_EQ(twoRemotingCommand.getOpaque(), 0);
+    EXPECT_EQ(twoRemotingCommand.getRemark(), "");
+    EXPECT_EQ(twoRemotingCommand.getVersion(), MQVersion::s_CurrentVersion);
+    // EXPECT_EQ(twoRemotingCommand.getFlag() , 0);
+    EXPECT_EQ(twoRemotingCommand.getMsgBody(), "");
+    EXPECT_TRUE(twoRemotingCommand.getCommandHeader() == nullptr);
+
+    RemotingCommand threeRemotingCommand(13, new GetRouteInfoRequestHeader("topic"));
+    EXPECT_FALSE(threeRemotingCommand.getCommandHeader() == nullptr);
+
+    RemotingCommand frouRemotingCommand(13, "CPP", MQVersion::s_CurrentVersion, 12, 3, "remark",
```

Comment:
where to delete (new GetRouteInfoRequestHeader) ?

Reply:
RemotingCommand  destruct is delete GetRouteInfoRequestHeader  pointer

Reply:
RemotingCommand::~RemotingCommand() { m_pExtHeader = NULL; }'

destruct function will not delete the pointer(m_pExtHeader )